internal/rbd/encryption.go

```diff
@@ -135,8 +135,11 @@ func (ri *rbdImage) copyEncryptionConfig(cp *rbdImage) error {
 		return fmt.Errorf("failed to fetch passphrase for %q: %w",
 			ri, err)
 	}
-
-	cp.encryption, err = util.NewVolumeEncryption(ri.encryption.GetID(), ri.encryption.KMS)
+	if copyOnlyPassphrase {
+		cp.encryption, err = util.NewVolumeEncryption(cp.encryption.GetID(), cp.encryption.KMS)
```

Comment:
this seems wrong, setting `cp.encryption` based on the contents of `cp.encryption` is weird. It would need at minimum a very clear comment explaining why this is needed, and works at all.

Reply:
>this seems wrong, setting cp.encryption based on the contents of cp.encryption is weird. It would need at minimum a very clear comment explaining why this is needed, and works at all.


yeah, it was redundant :facepalm: , fixed now, PTAL
destination rbdvol.encryption for snapshot restore PVC will be set/Initialized earlier, so its not required